src/bci_build/package/basecontainers.py

```diff
@@ -84,42 +84,83 @@
     for os_version in ALL_BASE_OS_VERSIONS
 ]
 
+_FIPS_ASSET_BASEURL = "https://api.opensuse.org/public/build/"
+
 # https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp3991.pdf
 # Chapter 9.1 Crypto Officer Guidance
-_FIPS_15_SP2_ASSET_BASEURL = "https://api.opensuse.org/public/build/"
-_FIPS_15_SP2_BINARIES = [
+_FIPS_15_SP2_BINARIES: list[str] = [
     f"SUSE:SLE-15-SP2:Update/pool/x86_64/openssl-1_1.18804/{name}-1.1.1d-11.20.1.x86_64.rpm"
     for name in ("openssl-1_1", "libopenssl1_1", "libopenssl1_1-hmac")
 ] + [
     f"SUSE:SLE-15-SP1:Update/pool/x86_64/libgcrypt.15117/{name}-1.8.2-8.36.1.x86_64.rpm"
     for name in ("libgcrypt20", "libgcrypt20-hmac")
 ]
 
+# submitted, not yet certified
+_FIPS_15_SP4_BINARIES: list[str] = [
+    f"SUSE:SLE-15-SP4:Update/pool/x86_64/openssl-1_1.28168/{name}-1.1.1l-150400.7.28.1.x86_64.rpm"
+    for name in ("openssl-1_1", "libopenssl1_1", "libopenssl1_1-hmac")
+] + [
+    f"SUSE:SLE-15-SP4:Update/pool/x86_64/libgcrypt.28151/{name}-1.9.4-150400.6.8.1.x86_64.rpm"
+    for name in ("libgcrypt20", "libgcrypt20-hmac")
+]
+
+# see grep -c -E "^ *V3.*key ID 39db7c82: OK" *.rpm | wc -l  below
+assert len(_FIPS_15_SP2_BINARIES) == len(_FIPS_15_SP4_BINARIES)
+
+
+def _get_fips_base_custom_end(os_version: OsVersion) -> str:
+    match os_version:
+        case OsVersion.SP3:
+            bins: list[str] = _FIPS_15_SP2_BINARIES
+        case OsVersion.SP4:
+            bins: list[str] = _FIPS_15_SP4_BINARIES
+        case _:
+            raise NotImplementedError(f"Unsupported os_version: {os_version}")
+    return "".join(
+        f"#!RemoteAssetUrl: {_FIPS_ASSET_BASEURL}{binary}\nCOPY {os.path.basename(binary)} .\n"
+        for binary in bins
+    ).strip()
+
+
+def _get_fips_pretty_name(os_version: OsVersion) -> str:
+    match os_version:
+        case OsVersion.SP3:
+            return f"{os_version.pretty_os_version_no_dash} FIPS-140-2"
+        case OsVersion.SP4:
+            return f"{os_version.pretty_os_version_no_dash} FIPS-140-3"
+        case _:
+            raise NotImplementedError(f"Unsupported os_version: {os_version}")
+
+
 FIPS_BASE_CONTAINERS = [
     OsContainer(
         name="base-fips",
         package_name="base-fips-image",
         exclusive_arch=[Arch.X86_64],
         os_version=os_version,
         build_recipe_type=BuildType.DOCKER,
-        support_level=SupportLevel.L3,
-        supported_until=datetime.date(2025, 12, 31),
+        support_level=(
+            SupportLevel.L3 if os_version == OsVersion.SP3 else SupportLevel.TECHPREVIEW
```

Comment:
SP4 is not L3 supported?

Reply:
hmm, I didn't mark it as supported because it is not yet a certified fips container. we could call it supported though if people read the README properly that should be fine. no strong opinion

Reply:
changed to L3